suite/tests/api/ir_aarch64.c

```diff
@@ -536,55 +540,55 @@ test_fmov_scalar(void *dc)
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(2.0));
+                                         opnd_create_immed_float(2.0f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(-2.0));
+                                         opnd_create_immed_float(-2.0f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(3.5));
+                                         opnd_create_immed_float(3.5f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(4.25));
+                                         opnd_create_immed_float(4.25f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(1.125));
+                                         opnd_create_immed_float(1.125f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(-1.125));
+                                         opnd_create_immed_float(-1.125f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(0.25));
+                                         opnd_create_immed_float(0.25f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(-0.25));
+                                         opnd_create_immed_float(-0.25f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(1.9375));
+                                         opnd_create_immed_float(1.9375f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(0.2109375));
+                                         opnd_create_immed_float(0.2109375f));
     test_instr_encoding(dc, OP_fmov, instr);
 
     instr = INSTR_CREATE_fmov_scalar_imm(dc, opnd_create_reg(DR_REG_S0),
-                                         opnd_create_immed_float(31.0));
+                                         opnd_create_immed_float(31.0f));
```

Comment:
None of these need the new opnd_create_immed_double?  Is there any code that calls that new routine here as at least a sanity check?

Reply:
Good call. Added `double` tests which flushed out encoding and decoder errors.